packages/backend/src/auth/auth.service.ts

```diff
@@ -62,4 +68,89 @@ export class AuthService {
   async validateUser(payload: SignInPayload) {
     return await this.usersService.findByPayload(payload);
   }
-}
+
+  async resetPassword(email: string) {
+    try {
+
+      const user = await this.usersService.findByEmail(email);
+
+      if (!user) {
+        throw new HttpException(
+            ErrorMessage.USER_NOT_FOUND,
+            HttpStatus.NOT_FOUND,
+        );
+      }
+
+      const payload = { id: user.id, email: user.email };
+
+      const jwtToken = sign(payload, process.env.SECRET_KEY, {
+        expiresIn: process.env.EXPIRES_RESET,
+      });
+
+      const options = {
+        viewEngine: {
+          partialsDir: __dirname + "/views/partials",
+          layoutsDir: __dirname + "/views/layouts",
+          extname: ".hbs"
+        },
+        extName: ".hbs",
+        viewPath: __dirname + "/views"
+      };
+
+      const transporter = nodemailer.createTransport({
+        host: this.configService.get<string>('MAIL_HOST'),
+        port: Number(this.configService.get<number>('MAIL_PORT')),
+        secure: (this.configService.get<string>('MAIL_SECURE') === 'true'),
+        auth: {
+          user: this.configService.get<string>('MAIL_LOGIN'),
+          pass: this.configService.get<string>('MAIL_PASSWORD'),
+        },
+      });
+
+      transporter.use("compile", hbs(options));
+
+      await transporter.sendMail({
+        from: this.configService.get<string>('MAIL_SENDER'),
+        to: email,
+        subject: 'Reset Password',
+        template: "resetPassword",
+        context: {jwtToken : jwtToken}
+      });
+
+      return { message: ResponseMessage.EMAIL_SEND, jwtToken };
+
+    } catch (error) {
+      return { message: error.message ? error.message : 'Error', error };
+    }
+  }
+
+  async changePassword(changePasswordDTO: ChangePasswordDTO) {
```

Comment:
dzieje sie tu sporo poza zmiana hasła np. veryfikacja tokenu (ja bym to ograł własnym guardem), wyłuskiwanie danych z tokenu itp

Reply:
chciałbym przegadać bo wydaje mi się to nie potrzebne